src/APICaller.js

```diff
@@ -0,0 +1,21 @@
+async function fetchWeatherData(url) {
+  const response = await fetch(url, {
+    mode: 'cors',
```

Comment:
Since you expect json response, it's a good idea to send `'Content-Type': 'application/json'` in the header.

Reply:
```
async function fetchWeatherData(url) {
  const response = await fetch(url, {
    mode: 'cors',
    **headers: { 'content-type': 'application/json' },**
  });
```
su šitu header gaunu tokį errorą:

_Access to fetch at 'http://api.openweathermap.org/data/2.5/weather?q=Vilnius&APPID=9a40fda64112122250ff2786a779a184&units=metric' from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status._

ir nerandu kaip išspręst. Čia servas nepriima tokio headerio?